myosuite/envs/myo/assets/leg/MyoOSLController.py

```diff
@@ -0,0 +1,380 @@
+
+import numpy as np
+import copy
+
+from opensourceleg.control.state_machine import Event, State, StateMachine
+from opensourceleg.osl import OpenSourceLeg
+
+
+class MyoOSLStateMachine():
+
+    """
+    OSL Related
+    """
+    def __init__(self, 
+                 body_mass,
+                 init_state='e_stance',
+                 hardware_param=None,
+                 debug=False,
+                 ):
+        """
+        Initializes the OSL state machine
+        Default init state: early stance [e_stance]
+        All states: [e_stance, l_stance, e_swing, l_swing]
+        - Early stance, Late Stance, Early Swing, Late Swing
+        """
+        assert init_state in ['e_stance', 'l_stance', 'e_swing', 'l_swing'], "Phase should be : ['e_stance', 'l_stance', 'e_swing', 'l_swing']"
+
+        self.initDefaults()
+        self.BODY_MASS = body_mass
+        self.BODY_WEIGHT = self.BODY_MASS * self.GRAVITY
+        self.DEBUG_MODE = debug
+
+        if hardware_param is not None:
+            self.HARDWARE = hardware_param
+
+        self.OSL = OpenSourceLeg(frequency=200)
+        self.OSL.add_joint(name="knee", gear_ratio=self.HARDWARE['knee']['gear_ratio'], offline_mode=True)
+        self.OSL.add_joint(name="ankle", gear_ratio=self.HARDWARE['ankle']['gear_ratio'], offline_mode=True)
+
+        self.FSM = self.build_state_machine(init_state)
+
+    def start(self):
+        """
+        Starts the state machine
+        """
+        self.FSM.start()
+
+    def reset(self, init_state, body_mass=80):
+        """
+        Rebuilds the state machine with the specified initial state
+        """
+        if body_mass is not None:
+            self.BODY_MASS = body_mass
+            self.BODY_WEIGHT = self.BODY_MASS * self.GRAVITY
+        
+        self.FSM = self.build_state_machine(init_state)
+
+    def update(self, sens_data):
+        """
+        Updates sensory data for OSL leg
+        Expects a dictionary of sensor values
+        """
+        assert all([key in sens_data.keys() for key in ['knee_angle', 'knee_vel', 'load_cell', 'ankle_angle', 'ankle_vel']]), "Missing data, dictionary should contain all of these keys ['knee_angle', 'knee_vel', 'load_cell', 'ankle_angle', 'ankle_vel']"
+        self.SENSOR_DATA = sens_data
+
+        print()
+
+        self.FSM.update()
+
+    def get_torques(self):
```

Comment:
function doc string missing.

Reply:
Added doc strings for all functions in controller. Note that file name has been changed to myoosl_control.py